src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -293,7 +293,7 @@ protected virtual void ConfigureWebHost(IWebHostBuilder builder)
         /// <param name="handlers">A list of <see cref="DelegatingHandler"/> instances to set up on the
         /// <see cref="HttpClient"/>.</param>
         /// <returns>The <see cref="HttpClient"/>.</returns>
-        public HttpClient CreateDefaultClient(Uri baseAddress, params DelegatingHandler[] handlers)
+        public virtual HttpClient CreateDefaultClient(Uri baseAddress, params DelegatingHandler[] handlers)
```

Comment:
I'd suggest  the following instead. 

Add a new method to `protected virtual CreateClientCore(params DelegatingHandler[] handlers)`

`HttpClient` has a lot of different things to configure. Whichever method is the **most fundamental** can't bake in things like `baseAddress`.

These convenience overloads should look like:
```
public HttpClient CreateDefaultClient(Uri baseAddress, params DelegatingHandler[] handlers)
{
    var client = CreateClientCore(handlers);
    client.BaseAddress = baseAddress;
    return client;
}
```

You could pivot back and forth on whether or not the most fundamental method is `Create` or a `Configure`.

This way someone can define additional overload